experanto/datasets.py

```diff
@@ -0,0 +1,448 @@
+                assert mode in ['standardize', 'normalize']
+                means = np.load(self._experiment.devices[device_name].root_folder / "meta/means.npy")
+                stds = np.load(self._experiment.devices[device_name].root_folder / "meta/stds.npy")
+                if mode == 'standardize':
+                    # If modality should only be standarized, set means to 0.
+                    means = np.zeros_like(means)
+
+                self._statistics[device_name]["mean"] = means.reshape(1, -1)  # (n, 1) -> (1, n) for broadcasting in __get_item__
+                self._statistics[device_name]["std"] = stds.reshape(1, -1)  # same as above
+
+    def initialize_transforms(self):
+        transforms = {}
+        for device_name in self.device_names:
+            if device_name == "screen":
+                # for the screen, we'll utilize torchvision.transforms
+                add_channel = Lambda(lambda x: torch.from_numpy(x[:, None, ...]) if len(x.shape) == 3 else torch.from_numpy(x))
+                transform_list = [v for v in self.modality_config.screen.transforms.values()]
+                transform_list.insert(0, add_channel)
+                transforms[device_name] = Compose(transform_list)
+            else:
+                if self.modality_config[device_name].transforms:
+                    transforms[device_name] = Compose([
+                        ToTensor(),
+                        torchvision.transforms.Normalize(self._statistics[device_name]["mean"], self._statistics[device_name]["std"])])
+                else:
+                    transforms[device_name] = Compose([
+                        ToTensor()])
+        return transforms
+
+    def get_sample_in_meta_condition(self) -> dict:
+        """
+        iterates through all
+           for example:
+              if meta_conditions = {"tier": [train,train, ...], "stim_type": [type1, type2, ...]}
+              and valid_condition = {"tier": train, "stim_type": type2}
+           then the output would be {"tier": [True, True, ...], "stim_type": [False, True, ...]}
+        """
+        sample_in_meta_condition = {}
+        for k, v in self.modality_config["screen"]["valid_condition"].items():
+            sample_in_meta = []
+            for i, (condition, start, end) in enumerate(zip(self.meta_conditions[k], self._start_times, self._end_times)):
+                if condition == v or (condition == "blank" and self.modality_config["screen"]["include_blanks"]):
+                    sample_in_meta.append(
+                        (self._screen_sample_times >= start) & (self._screen_sample_times < end)
+                    )
+            sample_in_meta_condition[k] = np.stack(sample_in_meta).sum(0).astype(bool)
+        return sample_in_meta_condition
+
+    def get_full_valid_sample_times(self, ) -> Iterable:
+        """
+        iterates through all sample times and checks if they are in the meta condition of interest
```

Comment:
it took me a bit of time to get through it, so I would suggest to add a bit more info in the doc strings, see the suggestion below
```suggestion
        iterates through all sample times and checks if they could be used as the start times, eg if the next `self.chunk_sizes["screen"]` points are still valid based on the previous meta condition filtering
```

Reply:
I added more docstrings now